### PR TITLE
unattended access in epsat

### PR DESCRIPTION
**Pull request by dan-jones-iw**

### CODE REVIEW

File: tool/site/client-react/src/index.tsx [hunk 1]
```diff
@@ -28,6 +29,9 @@ async function startApplication(baseUrl: string): Promise<void> {
         <PageContainer>
           <BrowserRouter>
             <Switch>
+              <Route path={`${baseUrl}change-auth`}>
+                <ChangeAuthPage/>
+              </Route>
```
Comment (booshibooshi) [inline tool/site/client-react/src/index.tsx:34]: The original flow redirected to a default oauth login and once you were logged in the `change-auth` concept made sense, think it's lost meaning now that the redirect to change-auth is the default flow on initial visit to the site, `login` probably makes more sense than `change-auth` in the new context of the standard login flow

Comment (dan-jones-iw) [reply tool/site/client-react/src/index.tsx:34]: yea we thought people would want to test the difference in functionality between user and system level access. so they will likely want to change-auth while already logged in. happy to change to login just would imply you should log out before logging in again (not really thought about log-out at all)

File: tests/e2e/pact/services/get-access-token.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import {createJWT} from "./jwt"
 import axios from "axios"
 import {URLSearchParams} from "url"
-import * as fs from "fs"
+import fs from "fs"
 import path from "path"
 
 /* TODO - get from ADO / env variables */
```
Comment (booshibooshi) [inline tests/e2e/pact/services/get-access-token.ts:7]: Do we want to pickup this todo as part of this pr, or next?

File: tool/site/client-react/src/index.tsx (hunk 1 above)
Comment (booshibooshi) [reply tool/site/client-react/src/index.tsx:34]: yeah might be easier to make decisions once this is in int. future consideration maybe 👍

File: tool/site/client-react/src/pages/changeAuthPage.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  const response = await axiosInstance.post<AuthResponse>(
+    `${baseUrl}change-auth`,
+    {authMethod}
+  )
+  redirect(`${response.data.redirectUri}`)
+}
+
+const makeUnattendedLoginRequest = async (baseUrl: string) => {
+  const response = await axiosInstance.post<AuthResponse>(`${baseUrl}unattended-login`)
+
```
Comment (booshibooshi) [inline tool/site/client-react/src/pages/changeAuthPage.tsx:45]: newline 😉

File: tool/site/client-react/src/pages/changeAuthPage.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  const {baseUrl} = useContext(AppContext)
+
+  const [pressedUserButton, setPressedUserButton] = useState(false)
+
+  return <>
+    <Label isPageHeading>Change auth</Label>
+    {!pressedUserButton ? <>
+      <Label>Select access level:</Label>
+      <ButtonList>
+        <Button type="submit" onClick={() => setPressedUserButton(true)}>User</Button>
```
Comment (booshibooshi) [inline tool/site/client-react/src/pages/changeAuthPage.tsx:18]: maybe a rename to help context here? setPressedUserButton -> attendedAccessSelected

File: tool/site/server_hapi/src/routes/auth/get-unattended-access-token.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      return responseToolkit.response({e}).code(500)
+    }
+  }
+}
+
+interface TokenResponse {
+  access_token: string
+  expires_in: string
+  token_type: "Bearer"
+}
```
Comment (booshibooshi) [inline tool/site/server_hapi/src/routes/auth/get-unattended-access-token.ts:60]: 👍

File: tests/e2e/pact/services/get-access-token.ts (hunk 2 above)
Comment (dan-jones-iw) [reply tests/e2e/pact/services/get-access-token.ts:7]: ideally neither lol its work that won't really help us with epsat so would rather not spend time on it. this is just a script to print a token to console, only works for internal-dev atm

File: tool/site/client-react/src/pages/changeAuthPage.tsx (hunk 4 above)
Comment (dan-jones-iw) [reply tool/site/client-react/src/pages/changeAuthPage.tsx:18]: yea seen this pattern elsewhere, will pick up

**Review by booshibooshi — APPROVED** [review]

**Pull request merged**